spec/rspec/core/world_spec.rb

```diff
@@ -22,6 +22,30 @@ module RSpec::Core
       end
     end
 
+    describe "#examples_count" do
+      before do
+        configuration.filter_run_including :foo => "bar"
+      end
+
+      it "returns examples count that includes shared examples" do
```

Comment:
Does the problem only manifest itself when using shared examples?  It's not clear to me what the connection between shared examples and a wrong count is...